### ISSUE TITLE
Defining a field on the prototype in a class field decorator makes the field unwritable

### ISSUE THREAD

**Issue by djrenren** [swc-project/swc#3342]
### Describe the bug

If in a class field decorator like so:

```
class Foo {
   @d
   member = 4
}
```

I define a property on `Foo`s prototype using: 
`Object.defineProperty(Foo.prototype, "member", {get() {...} set() {...}, ...}`

Then, that property definition will be overwritten by `_applyDecoratedDescriptor`, with a descriptor that includes `writable: false`. 

This behavior differs from `tsc` where the property is not overwritten. 

(note, this bug occurs whether `legacyDecorators` set to true or false. It doesn't work in either case)

This bug was originally found when using the [`lit`](https://lit.dev) library from Google and has been reduced to the following:

### Input code

```typescript
/* 
  Proxy any reads and writes to this member. We'll store the value at some private symbol
  and read or write to it when the member is accessed.
*/
const d = (proto: any, name) => {
  let self = proto.constructor;
  const key = typeof name === "symbol" ? Symbol() : `__${name}`;

  Object.defineProperty(self.prototype, name, {
    get(): any {
      return (this as { [key: string]: unknown })[key as string];
    },
    set(value: unknown) {
      const oldValue = (this as {} as { [key: string]: unknown })[
        name as string
      ];
      (this as {} as { [key: string]: unknown })[key as string] = value;
    },
    configurable: true,
    enumerable: true,
  });
};

class Test {
  @d
  member = 4;
}

console.log(new Test().member);
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": true,
      "loose": true
    },
    "target": "es2022"
  }
}
```

### Playground link

https://play.swc.rs/?version=1.2.133&code=H4sIAAAAAAAAA5VSy07DMBC85ytGCImkqtILp0YB%2FgAkEBwQat1k24Y6dmU7LVGVf2dtRxXlgrgk1u7sPNaeTZAAT0Z%2F9RCqhyFRWz7VOJrGkYXTcNvGoqV2RSbHG91ICeu0IW4QDkJ2BOFgdUvYm%2BYgHMH27UpLZvZMnhPaREZP2Dgct6TCfOQFK4iqImupzpPJLKm0sg41SqR7o52ee3dTKNFShvIOJ%2BaWxKok1wwKmDwMma5icwX3I8eOega4fk96HeZRliWuosMr3OM5nNIMcywXi%2BuTxwzLImGGx9UnVS6vad0o4iXtybg%2B9Zp5UPSs0dQ0OAI25NIsmB0L4PiuMwppWKOwOOGdPc15h6ZRm485OrVT%2BqgwZL7hIWOrCAzDNPwsM4dlnweys0RMqmX9Gm6j%2FCE2%2FCU5UiDu5iw%2BlkcP%2BA%2FjZQh2E2xfhGHD62bTGbGSnIfvjGKdVNfSr%2BqQFcnA11FJYS1eiJP63A81f8bXU%2BKWMUl4NVpSLvUmVXQM4DTLIyorvgHkLTWQ7QIAAA%3D%3D&config=H4sIAAAAAAAAAyWMQQqAMAwE%2F7JnD9JjfxNqFKXYko2glP7dirfZYZmGgwmxoYpR7SM%2Bp8uNCH%2BqMtleHROcQ62SqRMWTcXEixHR7Roml0L9Rx9fsU19FJRhDgG9v5KoSVNnAAAA

### Expected behavior

This code should log `4`.

In general, the decorator application logic for class fields should not define a property if a property is already defined.

### Actual behavior

This code fails to set the value of the field during object construction, because the property is marked as not writable and thus logs `undefined`.

### Version

1.2.133

### Additional context

This bug was likely introduced with the (admittedly kinda ironic PR): [fix(es/helpers): Fix decorators #3105](https://github.com/swc-project/swc/pull/3105/).

In typescript, class fields aren't defined as properties at all (at least by default). If you enable `useDefineForClassFields`, (which is more inline with the spec) you will *still* have a behavior discrepancy. `swc` will still log undefined, but `tsc` will log `4`. However, the property description from the decorator gets overwritten in the constructor with a more mundane:

```
Object.defineProperty(this, "member", {
  enumerable: true,
  configurable: true,
  writable: true,
  value: 4
});
```

meaning our custom setter and getter will never run.

Long story short though, this bug results in projects that build successfully with `tsc` but not with `swc` and thus should probably be fixed.
